### PR TITLE
[feat] generate turboOCI index from a gzip stream buffer

### PR DESCRIPTION
**Pull request by BigVan**
**What this PR does / why we need it**:

To make overlaybd can build a TurboOCI index from a stream buffer, 
which can make the building process of turboOCI faster.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [x]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by BigVan** [comment]
It is better to merge this PR after v1.0 has been released

File: src/overlaybd/gzindex/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 file(GLOB SOURCE_TAR "*.cpp")
-add_library(gzindex_lib STATIC ${SOURCE_TAR})
+add_library(gzindex_lib STATIC ${SOURCE_TAR} )
```
Comment (yuchen0cc) [inline src/overlaybd/gzindex/CMakeLists.txt:2]: nit: no need to modify

File: src/overlaybd/CMakeLists.txt [hunk 2]
```diff
@@ -7,6 +7,9 @@ add_subdirectory(extfs)
 add_subdirectory(gzip)
 add_subdirectory(gzindex)
 
+add_subdirectory(stream_convertor)
```
Comment (yuchen0cc) [inline src/overlaybd/CMakeLists.txt:10]: nit: no need alternate lines

File: src/overlaybd/gzindex/gzip_index_create.cpp [hunk 3]
```diff
@@ -13,13 +13,16 @@
    See the License for the specific language governing permissions and
    limitations under the License.
 */
+// #include "../zfile/crc32/crc32c.h"
```
Comment (yuchen0cc) [inline src/overlaybd/gzindex/gzip_index_create.cpp:16]: nit: remove?

File: src/overlaybd/stream_convertor/config.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+struct LogConfigPara : public App::ConfigGroup {
+    APPCFG_CLASS
+    APPCFG_PARA(level, uint32_t, 1);
+    APPCFG_PARA(path, std::string, "/var/log/overlaybd/stream-convertor.log");
+    APPCFG_PARA(limitSizeMB, uint32_t, 10);
+    APPCFG_PARA(rotateNum, int, 3);
+    APPCFG_PARA(mode, std::string, "stdout");
+};
+
+// struct ServerConfigPara : public App::ConfigGroup  {
```
Comment (yuchen0cc) [inline src/overlaybd/stream_convertor/config.h:33]: nit: remove?

File: src/overlaybd/tar/test/CMakeLists.txt [hunk 5]
```diff
@@ -4,10 +4,10 @@ link_directories($ENV{GFLAGS}/lib)
 include_directories($ENV{GTEST}/googletest/include)
 link_directories($ENV{GTEST}/lib)
 
-add_executable(untar_test test.cpp)
+add_executable(untar_test test.cpp )
```
Comment (yuchen0cc) [inline src/overlaybd/tar/test/CMakeLists.txt:7]: nit

File: src/overlaybd/tar/libtar.cpp [hunk 6]
```diff
@@ -75,12 +75,17 @@ int UnTar::set_file_perms(const char *filename) {
...
+        if (next == -1) {
+            return -1;
+        }
+        count++;
         if (TH_ISREG(header)) {
             auto size = get_size();
             file->lseek(((size + T_BLOCKSIZE - 1) / T_BLOCKSIZE) * T_BLOCKSIZE, SEEK_CUR); // skip size
         }
-        count++;
+        if (next != 0) break;
```
Comment (yuchen0cc) [inline src/overlaybd/tar/libtar.cpp:88]: why not test before TH_ISREG(header).
read_header ret: 0-normal, 1-eof, -1-error

**Review by yuchen0cc — CHANGES_REQUESTED** [review]

**Review by yuchen0cc — APPROVED** [review]
LGTM

**Pull request merged**